base-node/index.d.ts

```diff
@@ -131,6 +135,22 @@ export interface NodeOptions<Headers extends object = {}> {
    */
   inMap?: LogMapper
 
+  /**
+   * Function that will be called instead of adding action to the log
+   * after inMap and inFilter have been called.
+   *
+   * Use it if you want more control over when an action will be added to the log.
+   * ```js
+   * onActions(action, meta) {
+   *   // Add action to the log later
+   *   myActionQueue.schedule(() => {
+   *     log.add(action, meta)
+   *   })
+   * }
+   * ```
+   */
+  onActions?: ActionsCallback
```

Comment:
There are two problems with naming:

1. `onActions` has `s` on the end, but we call it for single action.
2. `log.add()` also add “action”, why we are not calling `onAction` there?

What do you think about `onSync`?